pkg/api/operators/apis/studyjob/v1alpha2/studyjob_types.go

```diff
@@ -0,0 +1,202 @@
+	// Type of job condition.
+	Type JobConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+}
+
+// JobConditionType defines all kinds of types of JobStatus.
+type JobConditionType string
+
+const (
+	JobCreated JobConditionType = "Created"
+	JobRunning JobConditionType = "Running"
+	JobRestarting JobConditionType = "Restarting"
+	JobSucceeded JobConditionType = "Succeeded"
+	JobFailed JobConditionType = "Failed"
+)
+
+// StudyJobStatus defines the observed state of StudyJob
+type StudyJobStatus struct {
+	// Represents time when the StudyJob was acknowledged by the StudyJob controller.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// Represents time when the StudyJob was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Represents last time when the StudyJob was reconciled. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	LastReconcileTime *metav1.Time `json:"lastReconcileTime,omitempty"`
+
+	Conditions               JobCondition  `json:"conditions,omitempty"`
+	StudyID                  string     `json:"studyId,omitempty"`
+	SuggestionParameterID    string     `json:"suggestionParameterId,omitempty"`
+	EarlyStoppingParameterID string     `json:"earlyStoppingParameterId,omitempty"`
+	Trials                   []Trial    `json:"trials,omitempty"`
+	BestObjectiveValue       *float64   `json:"bestObjectiveValue,omitempty"`
+	BestTrialID              string     `json:"bestTrialId,omitempty"`
+	BestWorkerID             string     `json:"bestWorkerId,omitempty"`
+	BestParameterValues      []ParameterValue    `json:"bestParameterValues,omitempty"`
+	CurrentSuggestionCount   int        `json:"currentSuggestionCount,omitempty"`
+}
+
+type WorkerCondition struct {
+	WorkerID       string      `json:"workerId,omitempty"`
+	Kind           string      `json:"kind,omitempty"`
+	Condition      Condition   `json:"condition,omitempty"`
+	ObjectiveValue *float64    `json:"objectiveValue,omitempty"`
+	StartTime      metav1.Time `json:"startTime,omitempty"`
+	CompletionTime metav1.Time `json:"completionTime,omitempty"`
+}
```

Comment:
How do you feel about allowing the option to add metadata to a trial? For example, when a trials fail we'll often add a message containing the error so we can figure out why that parameter combination didn't work (e.g. if it's an GPU OOM error then our model is too large and we might reduce the batch size or the number of neurons, if it's a loss=NaN error we reduce the maximum learning rate).